src/store/plugins/sdk.js

```diff
@@ -89,79 +94,78 @@ export default (store) => {
           ) {
             action.accept.apply(null, [
               ...(method === 'message.sign' ? [] : [null]),
-              { onAccount: { sign: () => {}, address: () => {} } },
+              { onAccount: { sign: () => { }, address: () => { } } },
             ]);
             return;
           }
           action.deny();
         };
 
-        sdk = await RpcWallet.compose({
-          methods: {
-            getApp(aeppUrl) {
-              return new App(aeppUrl);
-            },
-            async address(...args) {
-              const { address } = store.getters.account;
-              const app = args.pop();
-              if (
-                app instanceof App
-                && !(await store.dispatch('permissions/requestAddressForHost', {
-                  host: app.host.host,
-                  name: app.host.hostname,
-                  address,
-                  connectionPopupCb: async () => (IS_EXTENSION_BACKGROUND
-                    ? showPopup(app.host.href, 'connectConfirm')
-                    : store.dispatch('modals/open', {
-                      name: MODAL_CONFIRM_CONNECT,
-                      app: {
-                        name: app.host.hostname,
-                        icons: [],
-                        protocol: app.host.protocol,
-                        host: app.host.host,
-                      },
-                    })),
-                }))
-              ) return Promise.reject(new Error('Rejected by user'));
-              return address;
-            },
-            sign: (data) => (IS_EXTENSION_BACKGROUND
-              ? Crypto.sign(data, store.getters.account.secretKey)
-              : store.dispatch('accounts/sign', data)),
-            ...(IS_EXTENSION_BACKGROUND ? {} : {
-              signTransaction: (txBase64, opt) => (opt.onAccount
-                ? opt.onAccount.sign()
-                : store.dispatch('accounts/signTransaction', { txBase64, opt })),
-            }),
+        AeSdkWallet.prototype._resolveAccount = () => ({
+          address: store.getters.account.address,
+          sign: (unsigned) => (IS_EXTENSION_BACKGROUND
+            ? sign(unsigned, store.getters.account.secretKey)
+            : store.dispatch('accounts/sign', unsigned)),
+          signMessage: (unsigned) => (IS_EXTENSION_BACKGROUND
+            ? sign(unsigned, store.getters.account.secretKey)
+            : store.dispatch('accounts/sign', messageToHash(unsigned))),
+          ...(IS_EXTENSION_BACKGROUND ? {} : {
+            signTransaction: (txBase64, opt) => (store.dispatch('accounts/signTransaction', { txBase64, opt })),
+          }),
+          signTransaction: (txBase64, opt) => (store.dispatch('accounts/signTransaction', { txBase64, opt })),
+        });
+
+        AeSdkWallet.prototype.getAccounts = () => ({
```

Comment:
```suggestion
        class MyAeSdkWallet extends AeSdkWallet {
          getAccounts() {
```
will it work to do it in OOP way?

Reply:
Yes. I remember. I will work on it as soon as I get time. Kept the functional approach for now.

Reply:
Done